pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -882,7 +882,7 @@ protected BrokerResponse doHandleRequest(long requestId, String query, SqlNodeAn
         .filter(exception -> exception.getErrorCode() == QueryException.QUERY_VALIDATION_ERROR_CODE)
         .findFirst()
         .ifPresent(exception -> _brokerMetrics.addMeteredGlobalValue(BrokerMeter.QUERY_VALIDATION_EXCEPTIONS, 1));
-    if (QueryOptionsUtils.shouldDropResults(pinotQuery.getQueryOptions())) {
+    if (QueryOptionsUtils.shouldDropResults(pinotQuery.getQueryOptions()) && !pinotQuery.isExplain()) {
```

Comment:
(minor) Let's reverse these 2 checks because checking `explain` is much cheaper. Same for the other place